demos/Makefile

```diff
@@ -1,4 +1,4 @@
-MODULES=bio digest encode encrypt kdf keyexch mac pkey signature sslecho
+MODULES=cms digest encode encrypt kdf keyexch pkey signature sslecho
```

Comment:
Removing bio and mac seems wrong

Reply:
The "bio" and "mac" modules do not execute correctly since they need the vars `OPENSSL_INCS_LOCATION` and   `OPENSSL_LIBS_LOCATION` to be set.

Reply:
Could you just preset these to `../../include` and `../..` ?

Reply:
yes, I could certainly do that, but my main goal was to address #8026 not fix the demos :-)

Note that there are a number of demo modules not included in the variable `MODULES`.  e.g. `cipher` is not there.

Shane asked that I update the cms makefile so that `make test` works, which would be called from `demos/Makefile` if "cms" is added to `MODULES`.

Would it be okay if I just revert my changes to `demo/Makefile` in this PR?

I can open a new PR with changes to make `cd demos && make test` work.

Reply:
> Would it be okay if I just revert my changes to `demo/Makefile` in this PR?
> 
> I can open a new PR with changes to make `cd demos && make test` work.

+1 to that.

Reply:
I have just pushed that change.